addons/account/tests/test_account_account.py

```diff
@@ -456,3 +456,49 @@ def test_account_opening_balance(self):
             {'account_id': account.id,              'balance': 1000.0,  'amount_currency': 2000.0},
             {'account_id': account.id,              'balance': -1000.0, 'amount_currency': -2000.0},
         ])
+
+    def test_accoung_group_hierarchy_consistency(self):
+        """ Test if the hierarchy of account groups is consistent when creating, deleting and recreating an account group """
+        group_1 = self.env['account.group'].create({
```

Comment:
Might be cleaner to create a small inner function for creating account groups, this way you don't copy the code 5X

```python 
def create_account_group(name, code_prefix, company):
    return self.env['account.group'].create({
        'name': name,
        'code_prefix_start': code_prefix,
        'code_prefix_end': code_prefix,
        'company_id': company.id
    })